packages/gestalt/src/Link.js

```diff
@@ -18,6 +18,59 @@ import getRoundingClassName from './getRoundingClassName.js';
 import focusStyles from './Focus.css';
 import useFocusVisible from './useFocusVisible.js';
 import layoutStyles from './Layout.css';
+import Icon from './Icon.js';
+import Box from './Box.js';
+import Text from './Text.js';
+
+type ExternalLinkIcon =
+  | 'none'
+  | 'default'
+  | {|
+      size: $ElementType<React$ElementConfig<typeof Text>, 'size'>,
+      color: $ElementType<React$ElementConfig<typeof Icon>, 'color'>,
+    |};
+
+function ExternalIcon({
+  externalLinkIcon,
+  position,
+}: {|
+  externalLinkIcon: ExternalLinkIcon,
+  position: 'rtl' | 'ltr',
+|}): Node {
+  const externalLinkIconMap = {
```

Comment:
nit: this can be defined outside of the function so we're not creating a new object every render

Reply:
DONE